### PR TITLE
Add image layer for rotation operation

### PR DESCRIPTION
**Pull request by tnat410**
Rotate an image with an angle (in degree) around its center clockwise.
Using nearest neighbor pixel.

### CODE REVIEW

File: include/lbann/layers/image/rotation.hpp
Comment (timmoon10) [inline include/lbann/layers/image/rotation.hpp:107]:
```suggestion
    if (num_dims != 3) {
      err << get_type() << " layer \"" << this->get_name() << "\" "
          << "expects 3D input in CHW format, "
          << "but input dimensions are ";
      for (size_t i = 0; i < num_dims; ++i) {
        err << (i > 0 ? " x " : "") << dims[i];
      }
      LBANN_ERROR(err.str());
    } 
```

File: include/lbann/layers/image/rotation.hpp
Comment (timmoon10) [inline include/lbann/layers/image/rotation.hpp:116]: This isn't the right check. We expect `angle_dims` to be exactly equal to `{1}`.

File: src/layers/image/rotation.cpp
Comment (timmoon10) [inline src/layers/image/rotation.cpp:56]: We can simplify if we assume the data is in CHW format:
```suggestion
  const El::Int num_channels = input_dims[0]
  const El::Int input_height = input_dims[1];
  const El::Int input_width = input_dims[2];
```

File: src/layers/image/rotation.cpp
Comment (timmoon10) [inline src/layers/image/rotation.cpp:90]:
```suggestion
	  if((input_row >= 0 && input_row < input_height) && (input_col >= 0 && input_col < input_width)){
```

File: include/lbann/layers/image/rotation.hpp
Comment (timmoon10) [inline include/lbann/layers/image/rotation.hpp:78]: The base class implementation will automatically zero out the error signals:
```suggestion
```

**Review by timmoon10 — APPROVED** [review]
Overall looks good. [No new Bamboo failures](https://lc.llnl.gov/bamboo/browse/LBANN-TIM389-1).

We should figure out the LeNet hang before merging this. Also, next steps would be adding options to choose the output size, performing bilinear interpolation to reduce aliasing, and adding a GPU implementation.

File: include/lbann/layers/image/rotation.hpp
Comment (tnat410) [reply include/lbann/layers/image/rotation.hpp:116]: I think we should remove it. At first I thought we should not rotate with negative angle, but it should work with negative angle.

**Review by timmoon10 — CHANGES_REQUESTED** [review]
I'm holding this PR until we figure out the hang with multiple procs per trainer.

**Review by timmoon10 — APPROVED** [review]

**Pull request merged**
